x-pack/plugins/apm/server/lib/services/get_service_transaction_groups_metrics/index.ts

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Coordinate } from '../../../../typings/timeseries';
+import { EVENT_OUTCOME } from '../../../../common/elasticsearch_fieldnames';
+import { LatencyAggregationType } from '../../../../common/latency_aggregation_types';
+import { getLatencyValue } from '../../helpers/latency_aggregation_type';
+import { Setup, SetupTimeRange } from '../../helpers/setup_request';
+import { getTimeseriesDataForTransactionGroups } from '../get_service_transaction_groups/get_timeseries_data_for_transaction_groups';
+
+export async function getServiceTransactionGroupsMetrics({
+  serviceName,
+  transactionNames,
+  setup,
+  numBuckets,
+  searchAggregatedTransactions,
+  transactionType,
+  latencyAggregationType,
+}: {
+  serviceName: string;
+  transactionNames: string[];
+  setup: Setup & SetupTimeRange;
+  numBuckets: number;
+  searchAggregatedTransactions: boolean;
+  transactionType: string;
+  latencyAggregationType: LatencyAggregationType;
+}): Promise<
+  Record<
+    string,
+    {
+      latency: { timeseries: Coordinate[] };
+      throughput: { timeseries: Coordinate[] };
+      errorRate: { timeseries: Coordinate[] };
+    }
+  >
+> {
+  const { apmEventClient, start, end, esFilter } = setup;
+
+  const buckets = await getTimeseriesDataForTransactionGroups({
```

Comment:
IMHO this also can just be in this file, esp if we manage to simplify the code below